### PR TITLE
feat: changed API for login and registration

### PR DESCRIPTION
**Pull request by AleksandraSusel**

### CODE REVIEW

File: lib/data/data_source/authentication_remote_source_impl.dart [hunk 1]
```diff
@@ -11,61 +13,41 @@ import '../dto/user/create_user_dto.dart';
 
 @Injectable(as: AuthenticationRemoteSource)
 class AuthenticationRemoteSourceImpl implements AuthenticationRemoteSource {
-  AuthenticationRemoteSourceImpl(this.firebaseAuth);
+  AuthenticationRemoteSourceImpl(this.firebaseAuth, this.client, this.prefs);
 
   final FirebaseAuth firebaseAuth;
```
Comment (semiralharbi) [inline lib/data/data_source/authentication_remote_source_impl.dart:18]: Do you use this?

File: lib/data/data_source/authentication_remote_source_impl.dart [hunk 2]
```diff
@@ -11,61 +13,41 @@ import '../dto/user/create_user_dto.dart';
...
   }
 
+  Future<void> saveToken(String token) async {
+    await prefs.setString(SharedPrefsKeys.token, token);
+  }
+
+  Future<String?> getToken(LoginDto dto) async {
+    return prefs.getString('token');
+  }
+
```
Comment (semiralharbi) [inline lib/data/data_source/authentication_remote_source_impl.dart:42]: You can move this functions to SharedPrefsUtils so you can use the globally

File: lib/domain/repositories/authentication_repo.dart [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
 abstract class AuthenticationRepo {
-  Future<Either<Failure, UserCredential>> createUser(
-    CreateUserEntity entity,
-  );
-
   Future<Either<Failure, Success>> login(
     LoginEntity entity,
   );
 
+  Future<Either<Failure, Success>> createAccount(
```
Comment (semiralharbi) [inline lib/domain/repositories/authentication_repo.dart:13]: it's was good idea to change the name into `createAccount` to keep consistency with the endpoint, maybe you can also change the name of the entity?

File: lib/injectable/modules/api_module.dart [hunk 4]
```diff
@@ -2,10 +2,12 @@ import 'package:dio/dio.dart';
 import 'package:injectable/injectable.dart';
 import 'package:pretty_dio_logger/pretty_dio_logger.dart';
 
+import 'auth_interceptor.dart';
+
 @module
 abstract class ApiModule {
   @singleton
-  Dio get client => Dio()
+  Dio get client => Dio(BaseOptions(baseUrl: "http://192.168.8.151:4000/api/"))
```
Comment (semiralharbi) [inline lib/injectable/modules/api_module.dart:10]: base url should be fetched from local env variables like you did with the login data on drinks app

File: lib/injectable/modules/auth_interceptor.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import '../injectable.dart';
+
+class AuthInterceptor extends Interceptor {
+  AuthInterceptor();
+
+  @override
+  Future<void> onRequest(RequestOptions options, RequestInterceptorHandler handler) async {
+    final prefs = getIt<SharedPreferences>();
+    final token = prefs.getString('token');
```
Comment (semiralharbi) [inline lib/injectable/modules/auth_interceptor.dart:12]: use the SharedPrefsKeys

File: lib/presentation/pages/bloc/app_bloc.dart [hunk 6]
```diff
@@ -2,20 +2,26 @@ import 'package:equatable/equatable.dart';
...
-  AppBloc() : super(const AppState.loading()) {
+  AppBloc(this.prefs) : super(const AppState.loading()) {
     on<OnInitializeAppEvent>(_onInitializeAppEvent);
   }
 
+  final SharedPreferences prefs;
+
   Future<void> _onInitializeAppEvent(OnInitializeAppEvent event, Emitter<AppState> emit) async {
-    emit(const AppState.initial());
+    final token = prefs.get('token');
```
Comment (semiralharbi) [inline lib/presentation/pages/bloc/app_bloc.dart:20]: same

File: lib/presentation/pages/bloc/app_bloc.dart [hunk 7]
```diff
@@ -2,20 +2,26 @@ import 'package:equatable/equatable.dart';
...
+    if (token != null) {
+      emit(const AppState.success());
+    } else {
+      emit(const AppState.loading());
```
Comment (semiralharbi) [inline lib/presentation/pages/bloc/app_bloc.dart:24]: this should be loading?

File: test/data/data_source/authentication_data_source_test.dart [hunk 8]
```diff
@@ -1,150 +1,168 @@
...
-      await expectLater(
-        authenticationDataSource.createUser(dto),
-        throwsA(isA<ApiException>().having((e) => e.failure, 'Unknown Error', Errors.unknownError)),
-      );
-
-      verify(mockFirebaseAuth.createUserWithEmailAndPassword(email: mockedCreateUserDto.email, password: mockedCreateUserDto.password));
-      verifyNoMoreInteractions(mockFirebaseAuth);
-    },
-  );
+  // test(
```
Comment (semiralharbi) [inline test/data/data_source/authentication_data_source_test.dart:28]: do you need the commented tests?

**Review by semiralharbi — COMMENTED** [review]
Provide AppBloc tests

File: test/data/data_source/authentication_data_source_test.dart (hunk 8 above)
Comment (AleksandraSusel) [reply test/data/data_source/authentication_data_source_test.dart:28]: i left only sign out

**Review by semiralharbi — APPROVED** [review]

**Pull request merged**